src/client/datascience/notebook/notebookEditor.ts

```diff
@@ -95,6 +97,15 @@ export class NotebookEditor implements INotebookEditor {
         );
         disposables.push(model.onDidDispose(this._closed.fire.bind(this._closed, this)));
     }
+    @captureTelemetry(Telemetry.SyncAllCells)
+    public async syncAllCells(): Promise<void> {
+        // Need to read content out of the UI and save in our model
+        const modifiedCells = (this.model as VSCodeNotebookModel).getCells().map((c, i) => {
```

Comment:
Need to double check this. This is making an assumption that the cells in the document have the same relative position as they do in our notebook model

Reply:
Yeah they seem to.